### ISSUE TITLE
[Force deallocate] [CLI] To be part of regular deallocate operation in CLI

### ISSUE THREAD

**Issue by Aarthi-Vijaya-raghavan** [Azure/azure-cli#29808]
### Preconditions

- [X] No need to upgrade Python SDK or the Python SDK is ready.

### Resource Provider

microsoft.compute/virtual machines

### Description of Feature or Work Requested

ASK: deallocate VM is a operation that is most commonly used when VMs is not accessible due to any kind of failure. Deallocate depends on underlying layers to be available to be successful.  
With force deallocate customers will be able to deallocate their VM even during an outage. The ask is to provide force detach as one of the option along with regular deallocate.

Desing doc - https://microsoft.sharepoint.com/:w:/t/AzureIDC/AzureIDC_CRP/EX86dDdGWgdBsgo_yI6PuaEB3XGcDh-7Tm8rJT1NrQcqyA?e=3XnT5j

API: The development of the API is under progress with public preview planned in Nov 2024. GA in 2025 Q1.

### Minimum API Version Required

NA

### Swagger PR link / SDK link

Will share the swagger once API is deployed. 

### Target Date

2024-10-05

### PM Contact

aarthiv@microsoft.com

### Engineer Contact

asverma@microsoft.com

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by zhoxing-ms** [comment]
@Aarthi-Vijaya-raghavan May I ask if this feature request is related to security? Please note that currently, due to the CLI team being busy with the security wave and prioritizing security in this semester, so all the feature requests unrelated to security will be postponed. cc PM @Jacekey23

**Comment by Aarthi-Vijaya-raghavan** [comment]
@zhoxing-ms the feature is not related to security but is related to resiliency. The feature improves the resiliency of the VMs by allowing customers to deallocate their VM due to any failures and recover them within the same region/zone.

**Comment by zhoxing-ms** [comment]
@Aarthi-Vijaya-raghavan Since security is still the no.1 top things to catch up within Azure. but CLI will be in continous planning manner. If things can be justify as highly customer impact. please send the justification and we'll take it into consideration.

**Comment by Aarthi-Vijaya-raghavan** [comment]
@zhoxing-ms, filing this for Br semester. We are targeting this feature for public preview by Jul'25 and would need CLI support. The API side are being implemented and is expected to be ready by May'25. Please let me know how much time is required for CLI team to enable this feature once API is ready. Implementation wise this will be an optional parameter of "-force" in the existing deallocate/stop command.

**Comment by zhoxing-ms** [comment]
![Image](https://github.com/user-attachments/assets/00a4ff4f-e014-4c4c-9b4d-d1ea23c0d996)

It is currently estimated that the CLI team can complete this feature request by the end of June or early July. I think you need to fill in these necessary information before we can make a more accurate assessment

**Comment by Aarthi-Vijaya-raghavan** [comment]
Looping @***@***.***> and @***@***.***> for awareness.
We will reach out to you once we have the API swagger in place.


________________________________

**Comment by yanzhudd** [comment]
same as #29807
